src/Disks/VolumeJBOD.h

```diff
@@ -40,6 +44,16 @@ class VolumeJBOD : public IVolume
 
     /// Max size of reservation
     UInt64 max_data_part_size = 0;
+
+    bool areMergesAllowed() const;
+
+    void setAllowMergesFromQuery(bool allow);
+
+    /// True if parts on this volume participate in merges according to configuration.
+    bool are_merges_allowed_in_config = true;
```

Comment:
This option is suitable for all types of volumes, which will be inherited from `IVolume`. You should move it there.

Reply:
If I move that to `IVolume`, it will appear in `SingleDiskVolume` but it shall be stored in volume definition object, in order to update setting from configuration.

We shall either separate `IVolumeDefinition` from `IVolume` and place it there or just leave it as is. What would you recommend?

Reply:
This type of setting have to be accessible from all types of volumes. We can define struct `VolumeSettings` with `max_data_part_size` and `are_merges_allowed` (for now) and put a pointer to the settings object inside volumes. So settings object will be unique and volumes will contain pointers to this object. Original object will be created in `createVolume*` functions, which are now implemented in #11116. Maybe we should commit these refactorings and implement your functionality based on code from that PR.

Reply:
Moved volumes refactoring to separate PR #11883
Let's merge it before this one